### PR TITLE
chore: adds compiled css example

### PR DESCRIPTION
**Pull request by itsdouges**
Hiya! I was asked to add an example of how to use [Compiled](https://compiledcssinjs.com/) with Next.js, figured I might as well at it to the source 😄. Let me know if there's any changes needed.

### CODE REVIEW

**Review by github-Hesham — APPROVED** [review]

**Review by github-Hesham — APPROVED** [review]
heshamalhashmi #

**Review by github-Hesham — APPROVED** [review]
heshamalhashmi #

**Review by github-Hesham — DISMISSED** [review]
heshamalhashmi #

**Review by nadejdidjdhdhdh — APPROVED** [review]

**Review by marcodejongh — APPROVED** [review]

**Review by marcodejongh — DISMISSED** [review]
Just testing to see if randoms can approve PR's

File: examples/with-compiled-css/pages/class-names-box.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { ClassNames } from '@compiled/react'
```
Comment (andreipfeiffer) [inline examples/with-compiled-css/pages/class-names-box.js:1]: In Next.js, the files that are placed in the `/pages` folder become routes
A general practice is to put any included components in a separate root folder, like `/components`, so they don't become routes in the app

Only `index.js` should stay in `/pages` folder, as far as I can see, the others are only components

**Review by andreipfeiffer — CHANGES_REQUESTED** [review]

File: examples/with-compiled-css/.babelrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "presets": ["@babel/preset-react"],
```
Comment (andreipfeiffer) [inline examples/with-compiled-css/.babelrc:2]: I get `Error: Cannot find module '@babel/preset-react'`

Based on the official [Next.js Customizing Babel Config](https://nextjs.org/docs/advanced-features/customizing-babel-config) this should be:

```
{
  "presets": ["next/babel"],
  ...
}
```

**Review by andreipfeiffer — CHANGES_REQUESTED** [review]

File: examples/with-compiled-css/.babelrc [hunk 3]
```diff
@@ -1,4 +1,3 @@
 {
-  "presets": ["@babel/preset-react"],
-  "plugins": ["@babel/plugin-syntax-jsx"]
```
Comment (andreipfeiffer) [inline examples/with-compiled-css/.babelrc]: the additional plugin is required, for me it doesn't work without it, so `.babelrc` should look like:

```
{
  "presets": ["next/babel"],
  "plugins": ["@babel/plugin-syntax-jsx"]
}
```

**Review by andreipfeiffer — CHANGES_REQUESTED** [review]

File: examples/with-compiled-css/.babelrc [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["next/babel"]
+}
```
Comment (andreipfeiffer) [inline examples/with-compiled-css/.babelrc:3]: the additional plugin is required, for me it doesn't work without it, so `.babelrc` should look like:

```
{
  "presets": ["next/babel"],
  "plugins": ["@babel/plugin-syntax-jsx"]
}
```

**Review by andreipfeiffer — CHANGES_REQUESTED** [review]

**Review by andreipfeiffer — APPROVED** [review]

**Comment by andreipfeiffer** [comment]
@lfades @marcodejongh @timneutkens I have a question and can't seem to find an answer.

When you build this example, you **don't get a separate `chunk` for the library**, instead it's bundled inside each page.
The **same applies for** the examples for [Fela](https://github.com/vercel/next.js/tree/canary/examples/with-fela) and [JSS](https://github.com/vercel/next.js/tree/canary/examples/with-react-jss) and [Goober](https://github.com/vercel/next.js/tree/canary/examples/with-goober).

However, for other **CSS-in-JS** solutions like [Styled Components](https://github.com/vercel/next.js/tree/canary/examples/with-styled-components), [Emotion](https://github.com/vercel/next.js/tree/canary/examples/with-emotion), [Stitches](https://github.com/vercel/next.js/tree/canary/examples/with-stitches) or [TypeStyle](https://github.com/vercel/next.js/tree/canary/examples/with-typestyle) you **get a separate `chunk` of the lib**, that is shared across all pages.

So, the questions are:
- why is this difference? is there a reason for getting handled differently?
- is there a way to control what gets to common chunks, and what not? (I've searched all the official docs, without any luck)

I would really appreciate if you could shed some light.

I'm working on a [thorough analysis of all CSS-in-JS](https://github.com/andreipfeiffer/css-in-js) solutions with Next.js, and can't get my head around this tiny bit.

**Review by lfades — APPROVED** [review]
LGTM. Thank you @madou !

**Pull request merged**